### ISSUE TITLE
Change Cucumber's own features to use cucumber-expressions (from regexes)

### ISSUE THREAD

**Issue by mattwynne** [cucumber/cucumber-js#1871]
Right now we're using regular expressions in our own step definitions.

It would be good to change these over to using cucumber expressions (and parameter types!) so that we're providing a good example of how Cucumber should be used.

**Comment by davidjgoss** [comment]
Agreed!

**Comment by aslakhellesoy** [comment]
If you're mobbing with new contributors on this - show them https://cucumber.github.io/cucumber-expressions/ !

**Comment by shvmsnju** [comment]
Hi @mattwynne, I would like to contribute to this issue.

**Comment by mattwynne** [comment]
Awesome @shvmsnju, let me know if you need any help.

**Comment by pk** [comment]
Is this issues still valid/pending? I could have a look to get started with the code base.

**Comment by davidjgoss** [comment]
@pk absolutely, feel free to raise PRs and let me know if you need any pointers.

**Comment by pk** [comment]
Perfect, we are using Cucumber so much that we really would like to contribute. Also our test suites are getting more and more complex and we may need to start to use some features from API, running the Features dynamically so it would be nice to contribute before I start asking too many questions. ;)

**Comment by pk** [comment]
OK, I've looked into this and I think this task is done.

There are steps under `compability/**/*.ts` and `features/step_definitions/*.ts` all of which has been converted into the Cucumber Expressions.

I've done search for Given/When/Then with the RegExp patterns and it really yields references to the steps with RegExp only in *.md files.

I may be missing something....
